pkg/util/liveconfig/hive_test.go

```diff
@@ -6,69 +6,124 @@ package liveconfig
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	kcresp := &[]mgmtcontainerservice.CredentialResult{
+	kcresp := []*armcontainerservice.CredentialResult{
 		{
-			Name:  to.StringPtr("admin config"),
-			Value: to.ByteSlicePtr(kc),
+			Name:  to.Ptr("admin config"),
+			Value: kc,
 		},
 	}
 
-	resp := mgmtcontainerservice.CredentialResults{
+	resp := armcontainerservice.CredentialResults{
 		Kubeconfigs: kcresp,
 	}
 
-	mcc.EXPECT().ListClusterAdminCredentials(gomock.Any(), "rp-eastus", "aro-aks-cluster-001", "public").Return(resp, nil)
+	r := armcontainerservice.ManagedClustersClientListClusterAdminCredentialsResponse{
+		CredentialResults: resp,
+	}
+
+	transporter := fake.ManagedClustersServer{
+		NewListPager: func(options *armcontainerservice.ManagedClustersClientListOptions) (resp azfake.PagerResponder[armcontainerservice.ManagedClustersClientListResponse]) {
+			pagerResp := azfake.PagerResponder[armcontainerservice.ManagedClustersClientListResponse]{}
+			pagerResp.AddPage(http.StatusOK, armcontainerservice.ManagedClustersClientListResponse{
+				ManagedClusterListResult: armcontainerservice.ManagedClusterListResult{
+					Value: managedClustersList,
+				},
+			}, nil)
+
+			return pagerResp
+		},
+		ListClusterAdminCredentials: func(ctx context.Context, resourceGroupName, resourceName string, options *armcontainerservice.ManagedClustersClientListClusterAdminCredentialsOptions) (resp azfake.Responder[armcontainerservice.ManagedClustersClientListClusterAdminCredentialsResponse], errResp azfake.ErrorResponder) {
+			nilErr := azfake.ErrorResponder{}
+			rResp := azfake.Responder[armcontainerservice.ManagedClustersClientListClusterAdminCredentialsResponse]{}
+			rResp.SetResponse(http.StatusOK, r, nil)
+			return rResp, nilErr
+		},
+	}
+
+	mcc, err := utilcontainerservice.NewManagedClustersClientWithTransport(&azureclient.PublicCloud, dummySubscription, &azfake.TokenCredential{}, fake.NewManagedClustersServerTransport(&transporter))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	resultPage := mcc.List(ctx)
+
+	if resultPage == nil {
+		t.Fatal(errors.New("result page is nil"))
+	}
+
+	hasMore := resultPage.More()
+	if !hasMore {
+		t.Fatal(errors.New("result page has no results"))
+	}
+
+	results, err := resultPage.NextPage(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(results.Value) != len(managedClustersList) {
+		t.Fatal(errors.New("results page has an invalid number of results"))
+	}
+
+	if *results.Value[0].Name != *managedClustersList[0].Name {
+		t.Logf("expected %s to match %s", *results.Value[0].Name, *managedClustersList[0].Name)
+		t.Fatal(errors.New("expected name of first managed clusters list to match first page results"))
+	}
+
+	adminCredsResult, err := mcc.ListClusterAdminCredentials(ctx, "rp-eastus", "aro-ak-cluster-001")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(adminCredsResult.CredentialResults.Kubeconfigs) != 1 {
+		t.Fatal(errors.New("invalid number of credentials returned"))
+	}
```

Comment:
nit: Do we want to exit the test when it fails? or just `t.Fail` or `t.Error` will do?

https://pkg.go.dev/testing#T.Fail
https://pkg.go.dev/testing#T.Fatal

Reply:
`t.Fatal` in these cases. It's not written to continue checking things after a failure.